frontend/src/common/redux/userLogin.ts

```diff
@@ -208,40 +209,57 @@ export const registerViaEmailAndPassword = createAsyncThunk<
   {
     email: string;
     password: string;
-    displayName: string;
+    passwordConfirmation: string;
+    firstName: string;
+    lastName: string;
     recaptcha: string | null;
   },
   ThunkApiType
 >(
   "currentUser/registerViaEmailAndPassword",
-  async ({ email, password, displayName, recaptcha }, thunkAPI) => {
+  async ({ email, password, passwordConfirmation, firstName, lastName, recaptcha }, thunkAPI) => {
     if (!recaptcha) {
       return thunkAPI.rejectWithValue({
-        message: "Please complete the recaptcha",
+        message: "Please complete the recaptcha"
       });
     }
-    if (email === "") {
+    if (!email || email === "") {
       return thunkAPI.rejectWithValue({
-        message: "Please enter your email",
+        message: "Please enter your email"
       });
     }
-    if (password === "") {
+    if (!password || password === "") {
       return thunkAPI.rejectWithValue({
-        message: "Please enter your password",
+        message: "Please enter your password"
+      });
+    }
+
+    if (password !== passwordConfirmation) {
+      return thunkAPI.rejectWithValue({
+        message: "Passwords do not match"
       });
     }
-    if (displayName === "") {
+
+    if (!firstName || firstName === "" || !isNameValid(firstName)) {
+      return thunkAPI.rejectWithValue({
+        message: "Please enter your first name"
```

Comment:
we shold probably have a different message for if first name or last name is invalid since "please enter your first name" isn't descriptive/helpful to the user

Reply:
Should be fixed.

Reply:
code pointer?